impeller/display_list/dl_aiks_canvas.cc

```diff
@@ -0,0 +1,1191 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "impeller/display_list/dl_aiks_canvas.h"
+
+#include <algorithm>
+#include <cstring>
+#include <memory>
+#include <optional>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+#include "flutter/fml/logging.h"
+#include "flutter/fml/trace_event.h"
+#include "impeller/aiks/color_filter.h"
+#include "impeller/core/formats.h"
+#include "impeller/display_list/dl_image_impeller.h"
+#include "impeller/display_list/dl_vertices_geometry.h"
+#include "impeller/display_list/nine_patch_converter.h"
+#include "impeller/display_list/skia_conversions.h"
+#include "impeller/entity/contents/conical_gradient_contents.h"
+#include "impeller/entity/contents/filters/filter_contents.h"
+#include "impeller/entity/contents/filters/inputs/filter_input.h"
+#include "impeller/entity/contents/linear_gradient_contents.h"
+#include "impeller/entity/contents/radial_gradient_contents.h"
+#include "impeller/entity/contents/runtime_effect_contents.h"
+#include "impeller/entity/contents/scene_contents.h"
+#include "impeller/entity/contents/sweep_gradient_contents.h"
+#include "impeller/entity/contents/tiled_texture_contents.h"
+#include "impeller/entity/entity.h"
+#include "impeller/entity/geometry/geometry.h"
+#include "impeller/geometry/path.h"
+#include "impeller/geometry/path_builder.h"
+#include "impeller/geometry/scalar.h"
+#include "impeller/geometry/sigma.h"
+#include "impeller/typographer/backends/skia/text_frame_skia.h"
+
+namespace impeller {
+
+#define UNIMPLEMENTED \
```

Comment:
This is more or less the same implementation as dl_dispatcher right? if this sticks the landing we should clean that up quickly so they don't diverge.

Reply:
It's similar but a little different. The two major differences are:

- It can't really draw a display list (and vice versa, a display list can't draw an impeller picture)
- It's not using RTrees to determine culling when concatenating Impeller pictures (but it probably could if needed).

If this patch lands and sticks we can go ahead and delete dl_dispatcher, which is only used in tests. I didn't want to make this already large patch bigger by doing that now though.

Reply:
Sounds good to me

Reply:
I would rather...

- Delete this new class which is a poor approximation of nearly identical code in DisplayListBuilder
- Break DisplayListBuilder out into a preprocessor that already does the optimizations, and a pluggable backend which already exists and is called impeller::DlDispatcher
- Basically the preprocessor output can either go to a recorder which is "OpReceiver in, OpReceiver out" or directly to an impeller DlDispatcher.